### PR TITLE
Common metadata: Clarify in descriptions that the fields are not Item-only

### PR DESCRIPTION
**Pull request by m-mohr**
**Related Issue(s):** None

**Proposed Changes:**

1. Common metadata: Clarify in various field descriptions that the fields do not only apply to Items

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md)
      **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec),
      and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

**Review by constantinius — APPROVED** [review]
This clarifies the usage of common metadata fields to not only apply to Items. Is there perhaps a chance to put this document into a more central location (like `common/`)? Or at least add links from the catalog and collection specs?

**Comment by m-mohr** [comment]
I prefer the links from catalog/collection, as many other documents outside of the repo point to the file and we don't want to break these.

File: item-spec/common-metadata.md [hunk 1]
```diff
@@ -23,14 +23,14 @@ For example, `datetime` is required in STAC Items.
 
 ## Basics
 
-Descriptive fields to give a basic overview of a STAC Item.
+Descriptive fields to give a basic overview of a STAC resource (e.g. Catalog, Collection, Item, Asset).
```
Comment (philvarner) [inline item-spec/common-metadata.md:26]: I'd prefer the term `entity` or `object` instead of `resource`.

Comment (m-mohr) [reply item-spec/common-metadata.md:26]:
```suggestion
Descriptive fields to give a basic overview of a STAC entity (e.g. Catalog, Collection, Item, Asset).
```

File: item-spec/common-metadata.md [hunk 2]
```diff
@@ -46,11 +46,11 @@ Descriptive fields to give a basic overview of a STAC entity (e.g. Catalog, Coll
...
 
-| Field Name | Type         | Description                                                                      |
-| ---------- | ------------ | -------------------------------------------------------------------------------- |
-| datetime   | string\|null | See the [Item Spec Fields](item-spec.md#properties-object) for more information. |
+| Field Name | Type         | Description                                                                           |
+| ---------- | ------------ | ------------------------------------------------------------------------------------- |
+| datetime   | string\|null | See the [Item Spec Fields](item-spec.md#properties-object) for more information.      |
 | created    | string       | Creation date and time of the corresponding STAC entity or Asset (see below), in UTC. |
-| updated    | string       | Date and time the corresponding STAC entity or Asset (see below) was updated last, in UTC. |
+| u-----d    | s----g       | D-----------------------------------------------------------------------------------. |
```
Comment (m-mohr) [inline item-spec/common-metadata.md:53]: ??

Comment (emmanuelmathot) [reply item-spec/common-metadata.md:53]: oups. strange merge here...

**Review by emmanuelmathot — APPROVED** [review]

**Pull request merged**